src/transforms/glossary.ts

```diff
@@ -27,11 +55,35 @@ export function addGlossaryDefinition(
 
         if (at == null) return
 
+        const fragment: Descendant[] = Editor.fragment(editor, at)
+        const nodeToInsert: Node[] = []
+        const getToBottom = (descendant: Descendant[]) => {
+            if (Array.isArray(descendant)) {
+                descendant.some((elem: Descendant) => {
+                    if (VALID_DEFINITION_TERM_ELEMENTS.some(v => v(elem))) {
+                        if (Array.isArray(elem.children)) {
+                            JSON.parse(JSON.stringify(elem.children)).forEach((elem: Element) => {
+                                if (elem.type === "term") {
+                                    elem.children.forEach((item: Node) => {
+                                        nodeToInsert.push(item)
+                                    })
+                                } else {
+                                    nodeToInsert.push(elem)
+                                }
+                            })
+                        }
+                    }
+                    return getToBottom(elem.children as Descendant[])
+                })
+            }
+        }
```

Comment:
This should be a utility function, declared outside `addGlossaryDefinition`

I also don't understand why this filtering is necessary. Ensuring valid document structure should be done in normalizations.

Reply:
All this filtering is needed, because in some cases the text formatting should be preserved, in some cases not.

Reply:
I still don't understand why is it being done _here_, rather than in normalizations.

Reply:
I will move that to normalizations